api/v1alpha1/perconaservermysqlbackup_types.go

```diff
@@ -31,12 +31,14 @@ const (
 	BackupStarting  BackupState = "Starting"
 	BackupRunning   BackupState = "Running"
 	BackupFailed    BackupState = "Failed"
+	BackupError     BackupState = "Error"
```

Comment:
why don't we use `BackupFailed`?

Reply:
We can't use `BackupFailed` because of https://github.com/percona/percona-server-mysql-operator/pull/261/files#diff-ca9864e4a5a58ff77220d5b5b443428ef38a0e0ac4634c206c1e8b25621c3834L120.
I think that `Failed` state should mean that the running backup process failed. And `Error` should mean that the backup process didn't even start because of some reason that is possible to fix, and the backup will start working without the need to recreate the backup resource

Reply:
I see your point but please confirm this with @spron-in

Reply:
@pooknull can you pls check this with Sergey after which I think this can be merged.